back/sssscs/src/main/java/com/ib/user/UserServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.ib.user;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.ib.user.dto.UserCreateDto;
+import com.ib.user.exception.EmailTakenException;
+
+@Service
+public class UserServiceImpl implements UserService, UserDetailsService {
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+	
+	@Autowired
+	private IUserRepo userRepo;
+
+	@Override
+	public User register(UserCreateDto dto) {
+		if (isEmailTaken(dto.getEmail())) {
+			throw new EmailTakenException();
+		}
+		
+		User user = new User();
+		user.setEmail(dto.getEmail());
+		user.setPassword(passwordEncoder.encode(dto.getPassword()));
+		user.setName(dto.getName());
+		user.setSurname(dto.getSurname());
```

Comment:
Use modelMapper or automatic mapping for this. Maybe not necessarily right now, can do it later

Reply:
I'll do this on my branch